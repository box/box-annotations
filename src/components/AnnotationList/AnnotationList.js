// @flow
import React from 'react';

import Internationalize from '../Internationalize';
import Annotation from '../Annotation';

import './AnnotationList.scss';

type Props = {
    annotations: Annotations,
    language?: string,
    messages?: StringMap,
    onDelete: Function
};

const AnnotationList = ({ annotations, language, messages: intlMessages, onDelete }: Props) => (
    <Internationalize language={language} messages={intlMessages}>
        <ul className='ba-annotation-list'>
            {annotations.map(({ id, ...rest }) => (
                <li className='ba-annotation-list-item' key={`annotation_${id}`}>
                    <Annotation id={id} onDelete={onDelete} {...rest} />
                </li>
            ))}
        </ul>
    </Internationalize>
);

export default AnnotationList;
