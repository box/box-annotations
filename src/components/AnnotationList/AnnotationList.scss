@import '../../variables.scss';

.ba {
    .ba-annotation-list {
        padding-bottom: 10px;
        width: 100%;
    }

    .ba-annotation-list-item {
        list-style: none;

        // Spaces annotation elements from each other and the dividing border
        // Adds extra spacing above consecutive annotations as well
        margin-top: 23px; // Matches spacing of activity feed items in UI Elements
        padding-top: 15px;

        &:first-child {
            margin: 0;
        }
    }

    .ba-annotation-list-item + .ba-annotation-list-item {
        border-top: 1px solid $seesee;
    }
}
