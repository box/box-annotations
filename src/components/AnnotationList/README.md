### Description
Component to represent a list of annotations

### Examples
```jsx
const AnnotationListExample = require('examples').AnnotationListExample;

<AnnotationListExample />
```
