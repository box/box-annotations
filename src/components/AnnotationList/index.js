// @flow
export { default } from './AnnotationList';
