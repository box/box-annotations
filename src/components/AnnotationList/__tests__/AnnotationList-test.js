import * as React from 'react';
import { shallow } from 'enzyme';

import AnnotationList from '../AnnotationList';

const TIME_STRING_SEPT_27_2017 = '2017-09-27T10:40:41-07:00';

const USER = {
    type: 'user',
    id: '789ghi',
    name: 'Mia Thermopolis',
    email: 'princess@genovia.gov'
};

const annotations = [
    {
        id: '123',
        createdAt: TIME_STRING_SEPT_27_2017,
        createdBy: USER,
        message: 'test',
        permissions: {}
    },
    {
        id: '456',
        createdAt: TIME_STRING_SEPT_27_2017,
        createdBy: USER,
        message: 'test',
        permissions: {}
    }
];

const onDelete = () => console.log('deleted');

describe('components/AnnotationList', () => {
    test('should correctly render a list of annotation', () => {
        const wrapper = shallow(<AnnotationList annotations={annotations} onDelete={onDelete} />).dive();
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.ba-annotation-list-item').length).toEqual(2);
    });
});
