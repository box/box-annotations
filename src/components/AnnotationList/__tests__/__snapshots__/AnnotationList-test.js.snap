// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AnnotationList should correctly render a list of annotation 1`] = `
<IntlProvider
  locale="en"
  messages={Object {}}
>
  <ul
    className="ba-annotation-list"
  >
    <li
      className="ba-annotation-list-item"
      key="annotation_123"
    >
      <_class2
        createdAt="2017-09-27T10:40:41-07:00"
        createdBy={
          Object {
            "email": "princess@genovia.gov",
            "id": "789ghi",
            "name": "Mia Thermopolis",
            "type": "user",
          }
        }
        id="123"
        message="test"
        onDelete={[Function]}
        permissions={Object {}}
      />
    </li>
    <li
      className="ba-annotation-list-item"
      key="annotation_456"
    >
      <_class2
        createdAt="2017-09-27T10:40:41-07:00"
        createdBy={
          Object {
            "email": "princess@genovia.gov",
            "id": "789ghi",
            "name": "Mia Thermopolis",
            "type": "user",
          }
        }
        id="456"
        message="test"
        onDelete={[Function]}
        permissions={Object {}}
      />
    </li>
  </ul>
</IntlProvider>
`;
